### PR TITLE
chore: update manifests

### LINKED ISSUE #195: Update knative manfiests for `1.9.0-rc.2`
### Context

Each charm has a set of manifest files that have to be upgraded to their target version. The process of upgrading manifest files usually means going to the component’s upstream repository, comparing the charm’s manifest against the one in the repository and adding the missing bits in the charm’s manifest.

### What needs to get done

[See the release handbook](https://docs.google.com/document/d/1a4obWw98U_Ndx-ZKRoojLf4Cym8tFb_2S7dq5dtRQqs/edit?pli=1#heading=h.jt5e3qx0jypg)

### Definition of Done

Manifests and images are updated for the corresponding knative version used in kubeflow `1.9.0-rc.2`

### PR DESCRIPTION
**Pull request by mvlassis**
Closes #195 

This PR updates the manifest files of `knative-eventing`, `knative-serving`, `knative-operator`. No major changes were made.

### CODE REVIEW

File: charms/knative-eventing/config.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 options:
   version:
-    default: "1.12.3"
+    default: "1.12.6"
```
Comment (DnPlas) [inline charms/knative-eventing/config.yaml:6]: Should this be 1.12.4?

File: charms/knative-operator/src/manifests/auth_manifests.yaml.j2 [hunk 2]
```diff
@@ -11,7 +12,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# Source: knative/operator/config/rbac/clusterrole_aggregated.yaml
```
Comment (DnPlas) [inline charms/knative-operator/src/manifests/auth_manifests.yaml.j2:14]: Let's keep this comment as a reference for the future

**Review by DnPlas — CHANGES_REQUESTED** [review]
Few comments, other than that lgtm, thanks @mvlassis !

File: charms/knative-eventing/config.yaml (hunk 1 above)
Comment (mvlassis) [reply charms/knative-eventing/config.yaml:6]: This is according to the [Kubeflow component versions](https://github.com/kubeflow/manifests/tree/1.9.0-rc.2?tab=readme-ov-file#kubeflow-components-versions) section for 1.9.0-rc.2.

File: charms/knative-operator/src/manifests/auth_manifests.yaml.j2 (hunk 2 above)
Comment (mvlassis) [reply charms/knative-operator/src/manifests/auth_manifests.yaml.j2:14]: The comment has been kept, but moved to [to the top](https://github.com/canonical/knative-operators/pull/196/files#diff-c4921c7c33bf4dffd77abf51591560f4cfe9e1844f8f87375adbaf9156cc3c49R1) to be consistent with the other .yaml.j2 files.

File: charms/knative-eventing/config.yaml (hunk 1 above)
Comment (DnPlas) [reply charms/knative-eventing/config.yaml]: Ah, I did not notice! Thanks for the info (:

**Review by DnPlas — APPROVED** [review]
Thanks @mvlassis !

**Pull request merged**